hoomd/device.py

```diff
@@ -27,6 +27,50 @@
 from hoomd import _hoomd
 
 
+class NoticeFile:
+    """A file-like object that writes to a `hoomd.Device` notice stream.
+
+    Args:
+        device (`hoomd.Device`): The `Device` object.
+        level (int): Message notice level. Default value is 1.
+
+    Note:
+        Use this in combination with `Device.message_filename` to combine notice
+        messages with output from code that expects file-like objects (such as
+        `hoomd.write.Table`.
+    """
+
+    def __init__(self, device, level=1):
+        self._msg = device._cpp_msg
+        self._buff = ""
+        self._level = level
+
+    def write(self, message):
+        """Writes data to the associated devices notice stream.
+
+        Args:
+            message (str): Message to write.
+        """
+        if type(message) != str:
+            self._buff += str(message) + "\n"
+        else:
+            self._buff += message
```

Comment:
```suggestion
        self._buff += message
```

file-like objects expect `str` arguments. It is the responsibility of the caller to convert non-string arguments into strings with appropriate formatting.
```
>>> import sys
>>> sys.stdout.write(1234)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: write() argument must be str, not int
```

You can add an explicit type check and `raise TypeError`, or let the `+=` operator on `str` raise the error. Then you should remove the test that attempts to write an int.